### ISSUE TITLE
Add button to review page to add/remove add-on GUID to/from DeniedGuid

### ISSUE THREAD

**Issue by wagnerand** [mozilla/addons-server#13409]
Could we add a button to the reviewer tools to allow them to add/remove the add-on GUID to/from `DeniedGuid`?

If the add-on is deleted and the GUID is not in `DeniedGuid`, show a button "Deny resubmission".
If the add-on is deleted and the GUID is in `DeniedGuid`, show a button "Allow resubmission".
If the add-on is not deleted, do not show the button.

This should be gated by permission Reviews:Admin

Can we add an API for this as well? A client tool we use might want to add GUIDs to `DeniedGuid` in bulk. @kewisch might have ideas on what the API should look like exactly, for now I believe it would be something like:
* Check whether a single GUID is in `DeniedGuid`
* Add a single GUID to `DeniedGuid`
* Remove a single GUID from `DeniedGuid`

---

**For QA:** this is only about new buttons in the reviewer tools. These buttons are only available to admins and allow to deny/allow resubmission of an add-on (by adding the `guid` to the list of `DeniedGuid`). There is also activity log entries for each action.

Two new API endpoints have been added too. I'll link to the API docs when we have a link to share.

**Comment by willdurand** [comment]
> * Add a single GUID to `DeniedGuid`
> 
> * Remove a single GUID from `DeniedGuid`

I think this will be covered by the feature in the reviewer tools.

**Comment by willdurand** [comment]
> ~I think this will be covered by the feature in the reviewer tools.~ actually no, so I'll file a new issue for the API.

I filed https://github.com/mozilla/addons-server/issues/13424

**Comment by ioanarusiczki** [comment]
@willdurand I verified the new rev tools buttons on AMO dev with FF72(Win10).

When an add-on is deleted from admin tools by an admin from admin/models/addons/addon/ID/change/ the review page will display a red button "Deny resubmission" when the GUID is not in `DeniedGuid` or `Allow resubmission` when the GUID is in `DeniedGuid` for listed and unlisted add-ons. 

Examples:
https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/3roboform-password-manager
https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/515c5554a85b442e9c00

For `Allow resubmission` - the deleted add-on can be submitted again.
For `Deny resubmission` - a red warning is displayed by the linter -> `Duplicate add-on ID`.

My questions about the implementation :

- When the add-on is deleted from dev hub from the versions page, as an admin I will no longer see the review page, the page returns a 404 not found.  I don't know if it's expected (if the admin in production uses only the admin tools to delete the add-ons). 
-  I could not view the review pages for the deleted add-ons with a user added in the `Reviewers: Admins` group (which have the Reviews:Admin rule specified by @wagnerand ). I did not have this problem with the `Admin` group. 
- I did notice that for re-submitted add-ons the review page displays in the History section the previously deleted entries such as 
https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/207725b5b28d4f5a9b5b#history
but I can't tell about the activity log. Should an activity log about these actions be available in a queue from rev tools https://reviewers.addons-dev.allizom.org/en-US/reviewers/ ?
- Also shouldn't these review pages for deleted add-ons be available for the admins in a separate queue from rev tools, perhaps in Admin Tools section?

**Comment by wagnerand** [comment]
> When the add-on is deleted from dev hub from the versions page, as an admin I will no longer see the review page, the page returns a 404 not found. I don't know if it's expected (if the admin in production uses only the admin tools to delete the add-ons).

The review page should still be accessible by using the AMO numerical ID.

> I could not view the review pages for the deleted add-ons with a user added in the Reviewers: Admins group (which have the Reviews:Admin rule specified by @wagnerand ). I did not have this problem with the Admin group.

Yes, the mentioned permission is specifically to see that (and other) admin buttons. To access the reviewer tools in general, you need more permissions.

> I did notice that for re-submitted add-ons the review page displays in the History section the previously deleted entries such as
reviewers.addons-dev.allizom.org/en-US/reviewers/review/207725b5b28d4f5a9b5b#history
but I can't tell about the activity log. Should an activity log about these actions be available in a queue from rev tools reviewers.addons-dev.allizom.org/en-US/reviewers ?

If you mean the review log by activity log, then I don't think it should contain that action. The review log focuses on the common reviewer actions like approve, confirm approval, reject etc.

> Also shouldn't these review pages for deleted add-ons be available for the admins in a separate queue from rev tools, perhaps in Admin Tools section?

As mentioned above, admins should still be able to access them using the numerical AMO id (you might need `Addons:ViewDeleted` as an additional permission.

**Comment by ioanarusiczki** [comment]
@wagnerand Thanks for clarifications. I'm going to mark this verified fixed on AMO dev with FF72(Win10).
